test/livebook/session_test.exs

```diff
@@ -1906,6 +1919,36 @@ defmodule Livebook.SessionTest do
     assert_receive {:operation, {:add_cell_evaluation_output, _, ^cell_id, ^expected_output}}
   end
 
+  describe "default hub for new notebooks" do
+    test "use the default hub as default for new notebooks" do
+      hub = Livebook.Hubs.save_hub(@dummy_hub)
+      Livebook.Hubs.set_default_hub(hub.id)
+      notebook = Livebook.Session.default_notebook()
+
+      assert notebook.hub_id == hub.id
+      Livebook.Hubs.delete_hub(hub.id)
+    end
+
+    test "fallback to personal-hub when there's no default" do
+      hub = Livebook.Hubs.save_hub(@dummy_hub)
+      Livebook.Hubs.unset_default_hub(hub.id)
+      notebook = Livebook.Session.default_notebook()
+
+      assert notebook.hub_id == "personal-hub"
+      Livebook.Hubs.delete_hub(hub.id)
+    end
+
+    test "fallback to personal-hub when the default doesn't exist" do
+      hub = Livebook.Hubs.save_hub(@dummy_hub)
+      Livebook.Hubs.set_default_hub(hub.id)
+      Livebook.Hubs.delete_hub(hub.id)
+      notebook = Livebook.Session.default_notebook()
+
+      refute Livebook.Hubs.hub_exists?(hub.id)
+      assert notebook.hub_id == "personal-hub"
+    end
+  end
```

Comment:
We still want to wrap it in a module at the end, otherwise we run all tests synchronously, which we don't have to:

```elixir
  defmodule Global do
    use ExUnit.Case, async: false
    
    ...
  end
```